pkg/common/catalog/configure_posix.go

```diff
@@ -0,0 +1,26 @@
+//go:build !windows
+
+package catalog
+
+import (
+	"context"
+	"os"
+	"os/signal"
+
+	"golang.org/x/sys/unix"
+)
+
+func ReconfigureOnSignal(ctx context.Context, reconfigurer Reconfigurer) error {
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, unix.SIGUSR1)
+	defer signal.Stop(ch)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ch:
+			reconfigurer.Reconfigure(ctx)
```

Comment:
If I'm sending the SIGUSR1 signal to the process and I don't see any plugin being reconfigured (because I missed to configure `plugin_data_file`), I would probably benefit from having a log entry indicating that a SIGUSR1 signal was received.